pkg/assembler/clients/helpers/assembler.go

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helpers
+
+import (
+	"context"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/guacsec/guac/pkg/assembler"
+	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+)
+
+func GetAssembler(ctx context.Context, gqlclient graphql.Client) func([]assembler.AssemblerInput) error {
+
+	return func(preds []assembler.IngestPredicates) error {
+		for _, p := range preds {
+			if err := ingestCertifyScorecards(ctx, gqlclient, p.CertifyScorecard); err != nil {
```

Comment:
do we keep adding other ingestors here? Should there be a way to define which one runs so it doesn't try them all?

Reply:
yea we could add a check for len==0, which is kind of done in the function call through range, so not sure whether the compiler will optimize this out by unrolling the function...

Reply:
ah ok so you can add a check `len(p.CertifyScorecard) > 0` or skip it

Reply:
do you feel strongly for this to be in the main loop? since there is an equivalent check in the callee function?

Reply:
if the complier optimizes it all good